Kconfig.nrf

```diff
@@ -11,9 +11,23 @@ menu "Nordic nRF Connect"
 config BOARD_THINGY91_NRF9160NS
 	bool
 
-# nRF Connect SDK needs a larger default stacks for running certain tests.
+# Handle default to ensure that BT only will disable ENTROPY_CC3XX,
+# while multi protocol does not.
+config ENTROPY_CC3XX
+	default y if ZIGBEE || OPENTHREAD_NRF_SECURITY
+	default n if BT
+
+config ZIGBEE
+	select ENTROPY_CC3XX
+
+config OPENTHREAD_NRF_SECURITY
+	select ENTROPY_CC3XX
+
+# nRF Connect SDK needs a larger default stacks for running certain tests
+# or if ENTROPY_CC3XX is enabled.
 config MAIN_STACK_SIZE
 	default 2048 if ZTEST
+	default 2048 if ENTROPY_CC3XX && !(SPM || BUILD_WITH_TFM)
```

Comment:
What is the reason for adding `!(SPM || BUILD_WITH_TFM)` to this line? It is not explained in either the comment or the commit message. Also, these configs already select `ENTROPY_CC3XX ` by default.

Reply:
The SPM and BUILD_WITH_TFM will call into the Secure Partition Environment (secure domain in TrustZone). This means that the ENTROPY_CC3XX does not require having stack

Also note that in case of SPM / BUILD_WITH_TFM, the ENTROPY_CC3XX is enabled by default as the nRF9160 and nRF5340 does not contain a RNG device in the non-secure and would need to enable ENTROPY_CC3XX always

Reply:
This would apply to all stacks that call into the entropy driver, right? I think moving all CC3XX handling that requires this this stack increase to a thread so that stack increase would only have to be once.
We avoided using the chosen entropy device in the `bt_rand` function call because it would affect all BT threads because of this.

Reply:
We have started speaking about putting the entire entropy-driver in its own thread with a proficiently sized thread to ensure that this does not impact users. 

This, however, is not a quick fix and will not happen in the coming release of nRF Connect SDK